app/controllers/ems_common.rb

```diff
@@ -631,36 +633,76 @@ def deleteemss
+    if task == "refresh_ems"
+      model.refresh_ems(emss, true)
+      add_success_flash_and_audit(emss, task)
+    elsif task == "capture_ems"
+      model.where(:id => emss).each(&:queue_metrics_capture)
+      add_success_flash_and_audit(emss, task)
+    end
+
     return if @flash_array.present?
 
-    add_flash(n_("Refresh initiated for %{count} %{model} from the %{product} Database",
-                 "Refresh initiated for %{count} %{models} from the %{product} Database", emss.length) %
-      {:count   => emss.length,
-       :product => Vmdb::Appliance.PRODUCT_NAME,
-       :model   => ui_lookup(:table => table_name),
-       :models  => ui_lookup(:tables => table_name)})
+    add_flash(n_("%<action_name>s initiated for %<count>d %<model>s from the %<product>s Database",
+                 "%<action_name>s initiated for %<count>d %<models>s from the %<product>s Database", emss.length) %
+      {:action_name => action_name,
+       :count       => emss.length,
+       :product     => Vmdb::Appliance.PRODUCT_NAME,
+       :model       => ui_lookup(:table => table_name),
+       :models      => ui_lookup(:tables => table_name)})
+  end
+
+  def call_ems_refresh(emss)
+    call_ems_task(emss, "refresh_ems", _("Refresh"))
+  end
+
+  def call_ems_capture(emss)
+    call_ems_task(emss, "capture_ems", _("Metrics capture"))
+  end
+
+  def run_on_emss(action_method, action_name)
+    emss = find_checked_items
+    if emss.empty?
+      add_flash(
+        _("No %<model>s were selected for %<action_name>s") % {
+          :model       => ui_lookup(:table => table_name),
+          :action_name => action_name
+        },
+        :error,
+      )
+    end
+    send(action_method, emss)
+    show_list
+    @refresh_partial = "layouts/gtl"
+  end
+
+  def run_on_ems(action_method)
+    if params[:id].nil? || model.find_by(:id => params[:id]).nil?
+      add_flash(_("%{record} no longer exists") % {:record => ui_lookup(:table => table_name)}, :error)
+    else
+      send(action_method, [params[:id]])
+    end
+    params[:display] = @display
   end
 
   # Refresh VM states for all selected or single displayed ems(s)
   def refreshemss
     assert_privileges(params[:pressed])
     if @lastaction == "show_list"
-      emss = find_checked_items
-      if emss.empty?
-        add_flash(_("No %{model} were selected for refresh") % {:model => ui_lookup(:table => table_name)}, :error)
-      end
-      call_ems_refresh(emss)
-      show_list
-      @refresh_partial = "layouts/gtl"
+      run_on_emss(:call_ems_refresh, _("refresh"))
     else
-      if params[:id].nil? || model.find_by_id(params[:id]).nil?
-        add_flash(_("%{record} no longer exists") % {:record => ui_lookup(:table => table_name)}, :error)
-      else
-        call_ems_refresh([params[:id]])
-      end
-      params[:display] = @display
+      run_on_ems(:call_ems_refresh)
+    end
+  end
+
+  def captureemss
+    assert_privileges(params[:pressed])
+    if @lastaction == "show_list"
```

Comment:
Can you move this test into one method combining run_on_ems / run_on_emss?
Once you do that, you might be able to inline the real work to do (`refresh_ems`, `queue_metrics_capture`) right here, by having it take a block instead of a method name.

Reply:
see: https://github.com/ManageIQ/manageiq-ui-classic/pull/3188#discussion_r160595690